src/dune_rules/ml_sources.ml

```diff
@@ -294,14 +310,45 @@ let libs_and_exes dune_file ~dir ~scope ~lookup_vlib ~modules =
       | _ -> Memo.return List.Skip)
   >>| List.filter_partition_map ~f:Fun.id
 
+let melange_emits dune_file ~dir ~scope ~modules =
+  (* TODO: this function could be performed in parallel as part of libs_and_exes or some variation *)
```

Comment:
Yes, this should be done in 1 pass with inside libs_and_exes. In turn that function needs to be renamed to something like modules_to_stanzas.

Reply:
Fixed in https://github.com/ocaml/dune/pull/6349/commits/b5c8f5e15384e37ba6ce8e288a81602f9c0dd880.